System/ContentFile.cpp

```diff
@@ -342,4 +357,30 @@ namespace RTE {
 		}
 		return returnSample;
 	}
-}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ContentFile::ReloadBitmap(const std::string &filePath, int conversionMode) {
+		const int bitDepth = (conversionMode == COLORCONV_8_TO_32) ? BitDepths::ThirtyTwo : BitDepths::Eight;
+
+		std::unordered_map<std::string, BITMAP *>::iterator bmpItr = s_LoadedBitmaps[bitDepth].find(filePath);
+
+		if (bmpItr == s_LoadedBitmaps[bitDepth].end()) {
+			return;
+		}
+
+		PALETTE currentPalette;
+		get_palette(currentPalette);
+		set_color_conversion((conversionMode == COLORCONV_NONE) ? COLORCONV_MOST : conversionMode);
+
+		BITMAP *loadedBitmap = (*bmpItr).second;
+		BITMAP *newBitmap = load_bitmap(filePath.c_str(), currentPalette);
+		BITMAP swap;
+
+		std::memcpy(&swap, loadedBitmap, sizeof(BITMAP));
+		std::memcpy(loadedBitmap, newBitmap, sizeof(BITMAP));
+		std::memcpy(newBitmap, &swap, sizeof(BITMAP));
+
+		destroy_bitmap(newBitmap);
```

Comment:
Is it fine to rely on RAII to kill off `swap` here?
I've never looked into it, but I've always assumed there's some under-the-hood allegro work being done in `destroy_bitmap` that wouldn't be done if you `delete` the bitmap. If so, is there a problem if that work isn't being done for `swap`?